erpnext/selling/doctype/quotation/quotation.py

```diff
@@ -119,42 +119,42 @@ def update_lead(self):
 			doc.set_status(update=True)
 			doc.notify_update()
 
-	def update_opportunity(self):
-		for opportunity in list(set([d.prevdoc_docname for d in self.get("items")])):
-			if opportunity:
-				self.update_opportunity_status(opportunity)
-
-		if self.opportunity:
-			self.update_opportunity_status()
-
-	def update_opportunity_status(self, opportunity=None):
-		if not opportunity:
-			opportunity = self.opportunity
-
-		opp = frappe.get_doc("Opportunity", opportunity)
-		opp.set_status(update=True)
-		opp.notify_update()
+	def update_opportunity(self, reopen=False):
+		if self.get('opportunity'):
+			opp = frappe.get_doc("Opportunity", self.opportunity)
+			opp.set_status(update=True, status="Open" if reopen else None)
+			opp.notify_update()
 
 	def has_sales_order_or_invoice(self):
 		return frappe.db.get_value("Sales Order Item", {"quotation": self.name, "docstatus": 1})\
 			or frappe.db.get_value("Sales Invoice Item", {"quotation": self.name, "docstatus": 1})
 
-	def declare_enquiry_lost(self, lost_reasons_list, detailed_reason=None):
-		if not self.has_sales_order_or_invoice():
-			frappe.db.set(self, 'status', 'Lost')
+	@frappe.whitelist()
+	def set_is_lost(self, is_lost, lost_reasons_list=None, detailed_reason=None):
+		is_lost = cint(is_lost)
 
-			if detailed_reason:
-				frappe.db.set(self, 'order_lost_reason', detailed_reason)
+		if not is_lost and self.has_sales_order_or_invoice():
+			frappe.throw(_("Cannot set as Lost as Sales Order is made."))
```

Comment:
Improve Error message:
Cannot decare as Lost because Quotation has been converted to order